salt/states/user.py

```diff
@@ -143,6 +143,10 @@ def _changes(name,
         if expire and expire is not -1 and lshad['expire'] != expire:
             change['expire'] = expire
     # GECOS fields
+    if isinstance(fullname, str):
+        fullname.decode('utf-8')
```

Comment:
I am very hesitant to put this in. This may cause serious regressions for machines with non-UTF-8 encoding.

cc: @s0undt3ch and @jfindlay